app/src/org/commcare/activities/StandardHomeActivity.java

```diff
@@ -44,10 +57,15 @@
 
     private StandardHomeActivityUIController uiController;
 
+    private FlexibleAppUpdateController appUpdateController;
+    private static final String APP_UPDATE_NOTIFICATION = "app_update_notification";
+
     @Override
     public void onCreateSessionSafe(Bundle savedInstanceState) {
         super.onCreateSessionSafe(savedInstanceState);
         uiController.setupUI();
+        appUpdateController = AppUpdateControllerFactory.create(this::handleAppUpdate, getApplicationContext());
```

Comment:
we should move all the in app update code here to the `HomeScreenBaseActivity` in order to support all different versions of the home activity.